src/Chapter1.hs

```diff
@@ -641,7 +662,9 @@ You need to use recursion in this task. Feel free to return to it later, if you
 aren't ready for this boss yet!
 -}
 
-firstDigit n = error "firstDigit: Not implemented!"
+firstDigit :: Int -> Int
+firstDigit n =
+  if div n 10 == 0 then n else firstDigit (div n 10)
```

Comment:
I see that CI run was cancelled by timeout. This happened because the implementation of `firstDigit` hangs. You can verify it in GHCi:

```haskell
ghci> div (-1) 10
-1
```

So, your function hangs on negative numbers and you need to handle them separately.